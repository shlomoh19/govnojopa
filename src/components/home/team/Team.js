import React, { useState } from 'react'
import People from './People'
import InstagramIcon from '../../../UI/icons/InstagramIcon'
import YoutubeIcon from '../../../UI/icons/YoutubeIcon'
import FacebookIcon from '../../../UI/icons/FacebookIcon'

import './Team.scss'
import ArrowDownIcon from '../../../UI/icons/ArrowDownIcon'


export default function () {
    const [openTeam, setOpen] = useState(false)
    const openTeamHandler = () => setOpen(!openTeam)

    const playHandler = event => event.target.play()
    const pauseHandler = event => {
        event.target.currentTime = 0
        event.target.pause()
    } 
    

    const Founder = 'https://lbefree.com/storage/casting/team/Lilit_founder.mp4'
    const FounderImg = 'https://lbefree.com/storage/casting/team/images/Lilit_founder.jpg'

    return (
        <div className="main__team">
            {/*FOUNDER*/}
            <div className="team__founder">
                <div className="team__photo" >
                    <video
                        loop
                        className="video-team"
                        poster={FounderImg}
                        onMouseOver={playHandler}
                        onMouseOut={pauseHandler}
                        src={Founder}
                    >
                    </video>
                </div>
                <div className="team__desc">
                    <span className="team__item-position">FOUNDER</span>
                    <span className="team__item-name">Lilit Unanyan</span>
                    <div className="team__item-contacts">
                        <InstagramIcon href="https://www.instagram.com/casadelili/" />
                        <YoutubeIcon href="https://www.youtube.com/channel/UCAZFH1xkBWGn5EbbGuqYcGA" />
                        <FacebookIcon href="https://www.facebook.com/casadelilit" />
                    </div>
                </div>
            </div>

            {/*MOBILE MENU BUTTON*/}
            <div onClick={openTeamHandler} className="team__button-mobile">
                <span>{openTeam ? 'Close the Team' : 'Our Team'}</span>
                <ArrowDownIcon style={openTeam ? { transform: 'rotate(180deg)' } : null}/>
            </div>

            {/*TEAM*/}
            <div className={openTeam ? "team__items team-open" : "team__items"}>
                {
                    People.map(person => {
                        const { id, position, FIO, photo, video, webp } = person
                        return (
                            <div key={id} className="team__item">
                                {
                                    video
                                        ? <video
                                            
                                            loop
                                            className="team__photo"
                                            poster={photo}
                                            onMouseOver={playHandler}
                                            onMouseOut={pauseHandler}
                                            src={video}
                                        >
                                        </video>
                                        :<picture>
                                            <source srcset={webp} type="image/webp"/>
                                            <img className="team__photo" src={photo} alt="sorry bro :'(" />
                                        </picture>
                                }
                                <div className="team__desc">
                                    <span className="team__item-position">{position}</span>
                                    <span className="team__item-name">{FIO}</span>
                                </div>
                            </div>
                        )
                    })
                }
            </div>
        </div>
    )
}
