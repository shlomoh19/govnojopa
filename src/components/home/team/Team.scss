@import '../../../UI/style.scss';


.main__team {
  margin: 0 auto;
}
.team__item {
  align-self: center;
  justify-self: center;
  margin-bottom: 20px;
  width: 100%;
  text-align: center;
  font-size: 25px;
}
.team__founder {
  margin: 20px auto;
}
.team__item-instagram {
  width: 30px;
}
.team__item-youtube {
  width: 40px;
  margin-left: 20px;
}
.team__photo {
  position: relative;
  width: 300px;
  height: 300px;
  margin: 0 auto;
  border: 4px solid $grey;
  border-radius: 50%;
  .video-team {
    position: absolute;
    width: 300px;
    height: 300px;
    border-radius: 50%;
  }
}
.team__desc {
  margin-top: 10px;
  display: flex;
  flex-direction: column;
  text-align: center;
}
.team__items {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
}
.team__item-position {
  font-weight: 400;
  color: $blue;
}

.team__item-name {
  width: 60%;
  margin: 5px auto;
  padding: 0px 10px;
  text-transform: uppercase;
  border-radius: 8px;
  color: $black;
}
.team__button-mobile {
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: 25px;
  background: #e9e9e9;
  border-radius: 10px;
  padding-top: 5px;
  margin-top: 20px;
  cursor: pointer;
  display: none;
  transition: all .2s;
  margin-bottom: 20px;
  border: 1px solid transparent;
}

.team__item-contacts {
  display: flex;
  align-items: center;
  justify-content: center;
  a {
    margin-right: 5px;
    margin-left: 5px;
  }
}

.fa-angle-down {
  transition: all .2s;
}

.team__items.team-open {
  display: grid;
  .team__item {
    width: 150px;
    min-height: 200px;
    .team__photo {
      width: 120px;
      height: 120px;
    }
  }
}

@media (max-width: 1440px) {
  .team__photo {
    width: 180px;
    height: 180px;
    .video-team {
      width: 180px;
      height: 180px;
    }
  }
  .team__item {
    font-size: 14px;
  }
  .team__founder {
    font-size: 16px;
  }
}

@media (max-width: 1100px) {
  .team__photo {
    width: 150px;
    height: 150px;
    .video-team {
      width: 150px;
      height: 150px;
    }
  }
  .team__item {
    font-size: 14px;
  }
  .team__founder {
    font-size: 16px;
  }
}

@media (max-width: 750px) {
  .team__items {
    grid-template-columns: 1fr 1fr;
  }
}
@media (max-width: 500px) {
  .team__items {
    grid-template-columns: 1fr;
    margin: 0;
    text-align: center;
    display: none;
    .team__photo {
      width: 200px;
      height: 200px;
      .video-team {
        width: 200px;
        height: 200px;
      }
    }
  }
  .team__button-mobile {
    display: flex;
  }
}
