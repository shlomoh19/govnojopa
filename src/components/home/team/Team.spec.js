import React from 'react';
import { shallow } from 'enzyme';
import Team from './Team';
import toJson from 'enzyme-to-json';

describe('Team', () => {
    it("should render Team component", () => {
        const component = shallow(<Team />);
        const wrapper = component.find(".main__team");
        expect(wrapper.length).toBe(1);
    });

    it("should render Team snapshot", () => {
        const component = shallow(<Team />);
        expect(toJson(component)).toMatchSnapshot();
    });
});