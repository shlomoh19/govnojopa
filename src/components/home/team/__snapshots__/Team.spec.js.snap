// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team should render Team snapshot 1`] = `
<div
  className="main__team"
>
  <div
    className="team__founder"
  >
    <div
      className="team__photo"
    >
      <video
        className="video-team"
        loop={true}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        poster="https://lbefree.com/storage/casting/team/images/Lilit_founder.jpg"
        src="https://lbefree.com/storage/casting/team/Lilit_founder.mp4"
      />
    </div>
    <div
      className="team__desc"
    >
      <span
        className="team__item-position"
      >
        FOUNDER
      </span>
      <span
        className="team__item-name"
      >
        Lilit Unanyan
      </span>
      <div
        className="team__item-contacts"
      >
        <InstagramIcon
          href="https://www.instagram.com/casadelili/"
        />
        <YoutubeIcon
          href="https://www.youtube.com/channel/UCAZFH1xkBWGn5EbbGuqYcGA"
        />
        <FacebookIcon
          href="https://www.facebook.com/casadelilit"
        />
      </div>
    </div>
  </div>
  <div
    className="team__button-mobile"
    onClick={[Function]}
  >
    <span>
      Our Team
    </span>
    <ArrowDownIcon
      style={null}
    />
  </div>
  <div
    className="team__items"
  >
    <div
      className="team__item"
      key="6"
    >
      <video
        className="team__photo"
        loop={true}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        poster="https://lbefree.com/storage/casting/team/images/Salman_voin.jpg"
        src="Salman_voin.mp4"
      />
      <div
        className="team__desc"
      >
        <span
          className="team__item-position"
        >
          PARTNERSHIP MANAGER
        </span>
        <span
          className="team__item-name"
        >
          SALMAN AHACHAD
        </span>
      </div>
    </div>
    <div
      className="team__item"
      key="4"
    >
      <picture>
        <source
          srcset="https://lbefree.com/storage/casting/team/images/Jenya_programmer.webp"
          type="image/webp"
        />
        <img
          alt="sorry bro :'("
          className="team__photo lazyload"
          data-src="https://lbefree.com/storage/casting/team/images/Jenya_programmer.jpg"
          width="100%"
        />
      </picture>
      <div
        className="team__desc"
      >
        <span
          className="team__item-position"
        >
          PROGRAMMER
        </span>
        <span
          className="team__item-name"
        >
          EUGENE SLUGIN
        </span>
      </div>
    </div>
    <div
      className="team__item"
      key="1"
    >
      <video
        className="team__photo"
        loop={true}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        poster="https://lbefree.com/storage/casting/team/images/Vanya_programmer.jpg"
        src="Vanya_programmer.mp4"
      />
      <div
        className="team__desc"
      >
        <span
          className="team__item-position"
        >
          PROGRAMMER
        </span>
        <span
          className="team__item-name"
        >
          IVAN KHANIN
        </span>
      </div>
    </div>
    <div
      className="team__item"
      key="10"
    >
      <video
        className="team__photo"
        loop={true}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        poster="https://lbefree.com/storage/casting/team/images/Danil_programmer.jpg"
        src="Daniel_programmer.mp4"
      />
      <div
        className="team__desc"
      >
        <span
          className="team__item-position"
        >
          PROGRAMMER
        </span>
        <span
          className="team__item-name"
        >
          DANIEL RASSOMAKHIN
        </span>
      </div>
    </div>
    <div
      className="team__item"
      key="15"
    >
      <video
        className="team__photo"
        loop={true}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        poster="https://lbefree.com/storage/casting/team/images/Samir_prog.jpg"
        src="Samir_prog.mp4"
      />
      <div
        className="team__desc"
      >
        <span
          className="team__item-position"
        >
          PROGRAMMER & MEDIA MANAGER
        </span>
        <span
          className="team__item-name"
        >
          SAMIR El MOUSAOUI
        </span>
      </div>
    </div>
    <div
      className="team__item"
      key="12"
    >
      <video
        className="team__photo"
        loop={true}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        poster="https://lbefree.com/storage/casting/team/images/Kristina_pr.jpg"
        src="Kristina_pr.mp4"
      />
      <div
        className="team__desc"
      >
        <span
          className="team__item-position"
        >
          PR MANAGER
        </span>
        <span
          className="team__item-name"
        >
          KRISTINA SOSIPATROVA
        </span>
      </div>
    </div>
    <div
      className="team__item"
      key="11"
    >
      <video
        className="team__photo"
        loop={true}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        poster="https://lbefree.com/storage/casting/team/images/Olya_pr.jpg"
        src="Olya_pr.mp4"
      />
      <div
        className="team__desc"
      >
        <span
          className="team__item-position"
        >
          PR MANAGER
        </span>
        <span
          className="team__item-name"
        >
          OLGA KORNEEVA
        </span>
      </div>
    </div>
    <div
      className="team__item"
      key="7"
    >
      <video
        className="team__photo"
        loop={true}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        poster="https://lbefree.com/storage/casting/team/images/Harek_T34_85_film.jpg"
        src="Harek_T34_85_film.mp4"
      />
      <div
        className="team__desc"
      >
        <span
          className="team__item-position"
        >
          FILM DIRECTOR
        </span>
        <span
          className="team__item-name"
        >
          OLGA ZVEZDOGLIAD
        </span>
      </div>
    </div>
    <div
      className="team__item"
      key="9"
    >
      <video
        className="team__photo"
        loop={true}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        poster="https://lbefree.com/storage/casting/team/images/Marina_design.jpg"
        src="Marina_design.mp4"
      />
      <div
        className="team__desc"
      >
        <span
          className="team__item-position"
        >
          WEB DESIGNER
        </span>
        <span
          className="team__item-name"
        >
          MARINA BIBIK
        </span>
      </div>
    </div>
    <div
      className="team__item"
      key="2"
    >
      <picture>
        <source
          srcset="https://lbefree.com/storage/casting/team/images/Lisa_design.webp"
          type="image/webp"
        />
        <img
          alt="sorry bro :'("
          className="team__photo lazyload"
          data-src="https://lbefree.com/storage/casting/team/images/Lisa_design.jpg"
          width="100%"
        />
      </picture>
      <div
        className="team__desc"
      >
        <span
          className="team__item-position"
        >
          WEB DESIGNER
        </span>
        <span
          className="team__item-name"
        >
          ELIZAVETA ZINENKO
        </span>
      </div>
    </div>
    <div
      className="team__item"
      key="3"
    >
      <video
        className="team__photo"
        loop={true}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        poster="https://lbefree.com/storage/casting/team/images/Danil_art.jpg"
        src="Danil_art.MP4"
      />
      <div
        className="team__desc"
      >
        <span
          className="team__item-position"
        >
          ART DESIGNER
        </span>
        <span
          className="team__item-name"
        >
          DANILA MALYUTIN
        </span>
      </div>
    </div>
    <div
      className="team__item"
      key="8"
    >
      <video
        className="team__photo"
        loop={true}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        poster="https://lbefree.com/storage/casting/team/images/Igor_pacan.jpg"
        src="Igor_pacan.mp4"
      />
      <div
        className="team__desc"
      >
        <span
          className="team__item-position"
        >
          ACCOUNT MANAGER
        </span>
        <span
          className="team__item-name"
        >
          IGOR LYAKHOV
        </span>
      </div>
    </div>
  </div>
</div>
`;
